### PR TITLE
Update env version, add cost for previous added func, add tests, etc.

### PR DESCRIPTION
**Pull request by cici37**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
Update env version, add cost for previous added func, add tests, etc.
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by jpbetz** [comment]
We should probably merge this without the default compatibility version bump since it adds tests that should have been included in the 1.28 bump and use https://github.com/kubernetes/kubernetes/pull/122934 for the default compatibility version bump since it adds tests for the new CEL capabilities that are introduced in 1.29.

File: staging/src/k8s.io/apiserver/pkg/cel/environment/base.go [hunk 1]
```diff
@@ -43,7 +43,7 @@ import (
 // desirable because it means that CEL expressions are portable across a wider range
 // of Kubernetes versions.
 func DefaultCompatibilityVersion() *version.Version {
-	return version.MajorMinor(1, 28)
+	return version.MajorMinor(1, 29)
```
Comment (jpbetz) [inline staging/src/k8s.io/apiserver/pkg/cel/environment/base.go:46]: Remove this from this PR since it's testing capabilities added in 1.28?

Comment (cici37) [reply staging/src/k8s.io/apiserver/pkg/cel/environment/base.go:46]: I also added test in validation_test which require the version bump to be in place

**Comment by cici37** [comment]
I have cherry picked the commit for adding additional tests(https://github.com/kubernetes/kubernetes/pull/122934/commits/a324705ca98e81d07c139ac216cab07f0fd7e40f) into this PR.  /cc @jpbetz 
Thanks!

**Pull request merged**